container-search/src/main/java/com/yahoo/vespa/streamingvisitors/StreamingBackend.java

```diff
@@ -368,10 +377,95 @@ public VisitorSession createVisitorSession(VisitorParameters params) throws Pars
         }
 
         @Override
-        public Visitor createVisitor(Query query, String searchCluster, Route route, String schema, int traceLevelOverride) {
-            return new StreamingVisitor(query, searchCluster, route, schema, this, traceLevelOverride);
+        public Visitor createVisitor(Query query, Route route, Visitor.Context context) {
+            return new StreamingVisitor(query, route, this, context);
+        }
+
+    }
+
+
+    static class FillHitResult {
+        final boolean ok;
+        final String error;
+        FillHitResult(boolean ok) {
+            this(ok, null);
+        }
+        FillHitResult(boolean ok, String error) {
+            this.ok = ok;
+            this.error = error;
         }
+    }
 
+    private FillHitResult fillHit(FastHit hit, DocsumPacket packet, String summaryClass) {
+        if (packet != null) {
+            // System.err.println("fillHit from packet");
+            byte[] docsumdata = packet.getData();
+            if (docsumdata.length > 0) {
+                return new FillHitResult(true, decodeSummary(summaryClass, hit, docsumdata));
+            }
+            // System.err.println("no data");
+        } else {
+            // System.err.println("null packet");
+        }
+        return new FillHitResult(false);
+    }
+
+    static protected class FillHitsResult {
+        public final int skippedHits; // Number of hits not producing a summary.
+        public final String error; // Optional error message
+        FillHitsResult(int skippedHits, String error) {
+            this.skippedHits = skippedHits;
+            this.error = error;
+        }
+    }
+    /**
+     * Fills the hits.
+     *
+     * @return the number of hits that we did not return data for, and an optional error message.
+     *         when things are working normally we return 0.
+     */
+    protected FillHitsResult fillHits(Result result, DocsumPacket[] packets, String summaryClass) {
+        // System.err.println("fill result with packets: " + packets.length);
+        int skippedHits = 0;
+        String lastError = null;
+        int packetIndex = 0;
+        for (Hit hit : iterableHits(result)) {
+            // System.err.println("fill? hit " + hit);
+            if (hit instanceof FastHit fastHit && ! hit.isFilled(summaryClass)) {
+                // System.err.println("yes for " + summaryClass);
+
+                DocsumPacket docsum = packets[packetIndex];
+
+                packetIndex++;
```

Comment:
The mapping from hit to packet index is confusing. Why can't `summaryMap` be used directly?

Reply:
Alternatively remove the `! hit.isFilled(summaryClass)` check if it's guaranteed to always evaluate to `true`